pkg/controller/customresource/customresource.go

```diff
@@ -3,30 +3,43 @@ package customresource
 import (
 	"context"
 
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/statushandler"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
 	"go.uber.org/zap"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-// GetResource queries the Custom Resource 'request.NamespacedName' and populates the 'resource' pointer.
-// Note the logic: any reconcile result different from nil should be considered as "terminal" and will stop reconciliation
-// right away (the pointer will be empty). Otherwise the pointer 'resource' will always reference the existing resource
-func GetResource(client client.Client, request reconcile.Request, resource runtime.Object, log *zap.SugaredLogger) workflow.Result {
+// PrepareResource queries the Custom Resource 'request.NamespacedName' and populates the 'resource' pointer.
+// It also performs some updates to the 'status' field to indicate the start of reconciliation.
+func PrepareResource(client client.Client, request reconcile.Request, resource mdbv1.AtlasCustomResource, log *zap.SugaredLogger) (workflow.Result, *workflow.Context) {
```

Comment:
Now this function also returns the `workflow.Context` as it's initialized with the existing conditions from the CR (would be bad to have the conditions slice jumpting from  full to empty and back)

Reply:
as discussed - rename back to `getResource` and call `indicateStartOfReconciliation` explicitely